sunpy/net/jsoc/jsoc.py

```diff
@@ -34,32 +35,23 @@ class NotExportedError(Exception):
     pass
 
 
-class JSOCResponse(BaseQueryResponseTable):
-    def __init__(self, table=None, client=None):
-        """
-        table : `astropy.table.Table`
-        """
-        super().__init__(table, client)
-        self.query_args = getattr(table, 'query_args', None)
-        self.requests = getattr(table, 'requests', None)
-
-    def __getitem__(self, item):
-        ret = super().__getitem__(item)
-        ret.query_args = self.query_args
-        ret.client = self._client
-        warnings.warn("Downloading of sliced JSOC results is not supported. "
-                      "All the files present in the original response will be downloaded.",
-                      SunpyUserWarning)
-        return ret
+class JSOCResponse(QueryResponseTable):
+    query_args = TableAttribute()
+    display_keys = ['T_REC', 'TELESCOP', 'INSTRUME', 'WAVELNTH', 'CAR_ROT']
+    _original_num_rows = TableAttribute(default=None)
```

Comment:
Maybe add a quick comment explaining what this variable is doing?

Reply:
```suggestion
    # This variable is used to detect if the result has been sliced before it is passed 
    # to fetch and issue a warning to the user about not being able to post-filter JSOC searches.
    _original_num_rows = TableAttribute(default=None)
```